src/protocols/rdp/gdi.c

```diff
@@ -190,6 +193,45 @@ BOOL guac_rdp_gdi_desktop_resize(rdpContext* context) {
 
     guac_display_layer_close_raw(default_layer, current_context);
 
+    /* Make json string containing monitor information */
+    char json[JSON_BUFFER_SIZE];
+    int pos = 0;
+    pos += snprintf(json + pos, JSON_BUFFER_SIZE - pos, "{");
+
+    for (int i = 0; i < rdp_client->disp->monitors_count; i++) {
+
+        /* Skip monitors that have not been initialized yet */
+        if (rdp_client->disp->monitors[i].requested_width == 0 ||
+            rdp_client->disp->monitors[i].requested_height == 0) {
+            continue;
+        }
+
+        /* Append monitor information to JSON string */
+        pos += snprintf(json + pos, JSON_BUFFER_SIZE - pos,
```

Comment:
A small piece of my soul is severed and sent to the depths of Hades every time I see sprintf generating json.